更新最新源码后编译出错

ipq807x

Create index file './feeds/packages.index' 
Create index file './feeds/luci.index' 
Create index file './feeds/routing.index' 
Create index file './feeds/telephony.index' 
Create index file './feeds/oui.index' 
Create index file './feeds/helloworld.index' 
WARNING: Makefile 'package/feeds/packages/python-paho-mqtt/Makefile' has a dependency on 'python3-uuid', which does not exist
Installing all packages from feed packages.
Installing all packages from feed luci.
Installing all packages from feed routing.
Installing all packages from feed telephony.
Installing all packages from feed oui.
Installing all packages from feed helloworld.
这是缺少依赖？

Applying ./patches/001-procd_ujail_static_binary_fix.patch using plaintext: 
patching file jail/elf.c
Hunk #1 FAILED at 240.
1 out of 1 hunk FAILED -- saving rejects to file jail/elf.c.rej
Patch failed!  Please fix ./patches/001-procd_ujail_static_binary_fix.patch!
make[3]: *** [Makefile:181: /home/mitsuki/lede/build_dir/target-aarch64_cortex-a53_musl/procd-default/procd-2021-11-04-0ee8e734/.prepared_7809c874b685fcee9579ba572f1de4ed_6664517399ebbbc92a37c5bb081b5c53] Error 1
make[3]: Leaving directory '/home/mitsuki/lede/package/system/procd'
time: package/system/procd/default/compile#0.36#0.19#0.49
    ERROR: package/system/procd failed to build (build variant: default).
make[2]: *** [package/Makefile:116: package/system/procd/compile] Error 1
make[2]: Leaving directory '/home/mitsuki/lede'
make[1]: *** [package/Makefile:110: /home/mitsuki/lede/staging_dir/target-aarch64_cortex-a53_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory '/home/mitsuki/lede'
make: *** [/home/mitsuki/lede/include/toplevel.mk:230：world] 错误 2

--- a/package/system/procd/Makefile
+++ b/package/system/procd/Makefile
@@ -68,7 +68,7 @@ define Package/procd-ujail
   SECTION:=base
   CATEGORY:=Base system
   DEPENDS:=@KERNEL_NAMESPACES +@KERNEL_UTS_NS +@KERNEL_IPC_NS +@KERNEL_PID_NS \
-	  +libubox +libubus +libblobmsg-json
+	  +libubox +libubus +libuci +libblobmsg-json
   TITLE:=OpenWrt process jail helper
 endef
 

--- a/package/system/procd/patches/001-procd_ujail_static_binary_fix.patch
+++ b/package/system/procd/patches/001-procd_ujail_static_binary_fix.patch
@@ -1,16 +1,16 @@
 --- a/jail/elf.c
 +++ b/jail/elf.c
-@@ -240,18 +240,18 @@ int elf_load_deps(const char *path, cons
-
- 	gcc_mips64_bug_work_around = 1;
- #endif
+@@ -236,18 +236,18 @@ int elf_load_deps(const char *path, cons
+ 	unsigned long load_offset, load_vaddr;
+ 	unsigned long interp_offset;
+ 
 -	if (elf_find_section(map, PT_LOAD, &load_offset, NULL, &load_vaddr)) {
 -		ERROR("failed to load the .load section from %s\n", path);
 -		return -1;
 +	if (elf_find_section(map, PT_INTERP, &interp_offset, NULL, NULL) == 0) {
 +		add_path_and_deps(map+interp_offset, 1, -1, 0);
  	}
-
+ 
 -	if (elf_find_section(map, PT_DYNAMIC, &dyn_offset, &dyn_size, NULL)) {
 -		ERROR("failed to load the .dynamic section from %s\n", path);
 -		return -1;
@@ -18,12 +18,12 @@
 +		DEBUG("failed to load the .load section from %s\n", path);
 +		return 0;
  	}
-
+ 
 -	if (elf_find_section(map, PT_INTERP, &interp_offset, NULL, NULL) == 0) {
 -		add_path_and_deps(map+interp_offset, 1, -1, 0);
 +	if (elf_find_section(map, PT_DYNAMIC, &dyn_offset, &dyn_size, NULL)) {
 +		DEBUG("failed to load the .dynamic section from %s\n", path);
 +		return 0;
  	}
-
+ 
  	int clazz = map[EI_CLASS];

